### PR TITLE
Add ActionTaken message

### PR DESCRIPTION
**Pull request by djfarrelly**
### Purpose
To support the new actions taken middleware and tracking for new React-Redux applications in this PR: https://github.com/bufferapp/buffer-js-buffermetrics/pull/11

Project info: https://paper.dropbox.com/doc/Pitch-Automatic-Actions-Taken-AT2.0-BUDA-AKA3QrJsB3LJ4e2IGibw1

### Notes
```protobuf
// ActionTaken is an action that a user takes in an application directly, 
// not something done on their behalf or something done without direct user interaction
message ActionTaken {
    Uuid id = 1;

     // this is the top level scope, ex. "publish," "analyze," "extension"
    string application = 2;

    // this is the section of the application UI, ex. "queue," "composer," "billing"
    string location = 3; 

    // a unique action, ex. "profile_select," "share_now," "trial_cancel"
    string action = 4;
    // NOTE - Should we suggest these actions are of a certain tense? 

    // metadata - ex. profileId, userId - this replaces "extra_data" 
    map<string, string> metadata = 5; 
    google.protobuf.Timestamp created_at = 6;
}
```

### Review
Hey @davidgasquez and @michael-erasmus, I know Michael and I chatted on Slack about this new message type, but I wanted you too to take a look at it and give me feedback on it before publishing a new package!

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,4 +23,7 @@ clean-python:
 
 .PHONY: package-node
 package-node:
-	./scripts/package-node.sh
+	@./scripts/package-node.sh
+
+.PHONY: packages
```
Comment (davidgasquez) [inline Makefile]: 👌 Neat!

File: buda/entities/action_taken.proto [hunk 2]
```diff
@@ -0,0 +1,15 @@
+syntax = "proto3";
+
+package buda;
+
+import "buda/entities/uuid.proto";
+import "google/protobuf/timestamp.proto";
+
+message ActionTaken {
```
Comment (davidgasquez) [inline buda/entities/action_taken.proto]: Do you think we could add some comments from your _Notes_ section code in this file? I can see them being helpful for future reference.

Comment (djfarrelly) [reply buda/entities/action_taken.proto]: Good idea @davidgasquez! Maybe going off Michael's idea in #19, we can use a semi-standard way to document our messages? https://github.com/pseudomuto/protoc-gen-doc#writing-documentation

File: Makefile (hunk 1 above)
Comment (djfarrelly) [reply Makefile]: Happy to rename this one if you have a better name, this was just a placeholder for it for now 😄

File: buda/entities/action_taken.proto [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * ActionTaken is an action that a user takes in an application directly
+ */
+message ActionTaken {
+    Uuid id = 1;
+    // The application where the action is performed
+    string application = 2;
+    // The location in the application UI where the action is performed
+    string location = 3;
+    // A unique label for the action performed
+    string action = 4;
```
Comment (michael-erasmus) [inline buda/entities/action_taken.proto]: @djfarrelly great point about a convention on which tense to use. My hunch is past tense to stick with the ActionTaken tense? It's also a subtle hint that this is an immutable event

**Comment by michael-erasmus** [comment]
Hey @djfarrelly! I wanted to spend some time building out the Buda pipeline for ActionsTaken today. Curious if it feels ok to go ahead and merge this PR?

**Comment by michael-erasmus** [comment]
@djfarrelly went ahead and merged this. hope that's ok! 😸

**Pull request merged**